cmd/lifecycle-util.go

```diff
@@ -0,0 +1,327 @@
+	fieldTheme  string
+	columnWidth int
+	align       int
+}
+
+type showDetails struct {
+	allAvailable bool
+	expiry       bool
+	transition   bool
+	json         bool
+	minimum      bool
+}
+
+func getCentered(label string, maxLen int) string {
+	const toPadWith string = " "
+	lblLth := len(label)
+	if lblLth > 1 && lblLth%2 != 0 {
+		lblLth++
+	} else if lblLth == 1 {
+		lblLth = 2
+	}
+	length := (float64(maxLen - lblLth)) / float64(2)
+	rptLth := (int)(math.Floor(length / float64(len(toPadWith))))
+	leftRptLth := rptLth
+	rightRptLth := rptLth
+	if rptLth <= 0 {
+		leftRptLth = 1
+		rightRptLth = 0
+	}
+	output := strings.Repeat(toPadWith, leftRptLth) + label + strings.Repeat(toPadWith, rightRptLth)
+	return output
+}
+
+func getLeftAlgined(label string, maxLen int) string {
+	const toPadWith string = " "
+	lblLth := len(label)
+	length := maxLen - lblLth
+	if length <= 0 {
+		return label
+	}
+	output := strings.Repeat(toPadWith, 1) + label + strings.Repeat(toPadWith, length-1)
+	return output
+}
+
+func getRightAligned(label string, maxLen int) string {
+	const toPadWith string = " "
+	lblLth := len(label)
+	length := maxLen - lblLth
+	if length <= 0 {
+		return label
+	}
+	output := strings.Repeat(toPadWith, length) + label
+	return output
+}
+
+func getBucketNameFromURL(urlStr string) string {
+	clientURL := newClientURL(urlStr)
+	bucketName := splitStr(clientURL.String(), "/", 3)[1]
+	return bucketName
+}
+
+func getHostCfgFromURL(urlStr string) *hostConfigV9 {
+	clientURL := newClientURL(urlStr)
+	alias := splitStr(clientURL.String(), "/", 3)[0]
+	return mustGetHostConfig(alias)
+}
+
+func getParsedHostURL(urlStr string) (*url.URL, *probe.Error) {
+	hostCfg := getHostCfgFromURL(urlStr)
+	if hostCfg == nil {
+		return nil, errInvalidURL(urlStr)
+	}
+	hostURLParse, err := url.Parse(hostCfg.URL)
+	if err != nil {
+		return nil, probe.NewError(err)
+	}
+	return hostURLParse, nil
+}
+
+func getMinioClient(urlStr string) (*minio.Client, *probe.Error) {
+	var api *minio.Client
+	hostCfg := getHostCfgFromURL(urlStr)
+	if hostCfg == nil {
+		return nil, errInvalidTarget(urlStr)
+	}
+	creds := credentials.NewStaticV4(hostCfg.AccessKey, hostCfg.SecretKey, "")
+	options := minio.Options{
+		Creds:        creds,
+		Secure:       false,
+		Region:       "",
+		BucketLookup: minio.BucketLookupPath,
+	}
+	parsedHostURL, err := getParsedHostURL(urlStr)
+	if err != nil {
+		return nil, err
+	}
+	api, apierr := minio.NewWithOptions(parsedHostURL.Host, &options) // Used as hostname or host:port
+	if apierr != nil {
+		// fatalIf(probe.NewError(err), "Unable to connect `"+hostURLParse.Host+"` with obtained credentials.")]
+		return nil, probe.NewError(apierr)
+	}
+	return api, nil
+}
+
+func getIlmConfig(urlStr string) (lfcInfo lifecycleConfiguration, err error) {
+	var lfcInfoXML string
+	var pErr *probe.Error
+	if lfcInfoXML, pErr = getIlmInfo(urlStr); pErr != nil {
+		console.Errorln("Unable to get lifecycle configuration from:" + urlStr)
```

Comment:
you can use errorIf or fatalIf as that seems to be the convention in the rest of the mc code base. Unless there is a limitation in using errorIf/fatalIf